keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/ExceptionHandler.kt

```diff
@@ -77,6 +79,20 @@ class ExceptionHandler(
     return ApiError(e)
   }
 
+  @ExceptionHandler(AccessDeniedException::class)
+  @ResponseStatus(FORBIDDEN)
+  fun onAccessDenied(e: AccessDeniedException): ApiError {
+    log.error(e.message)
+    return ApiError(
+      if (e.message == null || e.message == "Access is denied") {
+        "Access denied. Please make sure you have access to the service account specified in your delivery config, " +
+          "and that the service account has access to the target application and cloud account."
+      } else {
+        e.message!!
```

Comment:
This allows messages that are _not_ the default (which may come from using the `AuthorizationSupport.checkXyZ()`-type functions) to be propagated.